internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -507,6 +510,113 @@ func (a *Adapter) EnsureOverrideSnapshotValid() (controller.OperationResult, err
+	if prGroup == "" {
+		a.logger.Error(fmt.Errorf("NotFound"), fmt.Sprintf("Failed to get PR group from snapshot %s/%s", a.snapshot.Namespace, a.snapshot.Name))
+		err := gitops.AnnotateSnapshot(a.context, a.snapshot, gitops.PRGroupCreationAnnotation, fmt.Sprintf("Failed to get PR group from snapshot %s/%s", a.snapshot.Namespace, a.snapshot.Name), a.client)
+		if err != nil {
+			return controller.RequeueWithError(err)
+		}
+		return controller.ContinueProcessing()
+	}
+
+	pipelineRuns, err := a.loader.GetPipelineRunsWithPRGroupHash(a.context, a.client, a.snapshot, prGroupHash)
+	if err != nil {
+		a.logger.Error(err, fmt.Sprintf("Failed to get build pipelineRuns for given pr group hash %s", prGroupHash))
+		return controller.RequeueWithError(err)
+	}
+
+	for _, pipelineRun := range *pipelineRuns {
+		pipelineRun := pipelineRun
+
+		// check if build PLR finishes
+		if !h.HasPipelineRunFinished(&pipelineRun) {
+			a.logger.Info(fmt.Sprintf("The build pipelineRun %s/%s with pr group %s is still running, won't create group snapshot", pipelineRun.Namespace, pipelineRun.Name, prGroup))
+			err := gitops.AnnotateSnapshot(a.context, a.snapshot, gitops.PRGroupCreationAnnotation, fmt.Sprintf("The build pipelineRun %s/%s with pr group %s is still running, won't create group snapshot", pipelineRun.Namespace, pipelineRun.Name, prGroup), a.client)
+			if err != nil {
+				return controller.RequeueWithError(err)
+			}
+			return controller.ContinueProcessing()
+		}
+
+		// check if build PLR succeeds
+		if !h.HasPipelineRunSucceeded(&pipelineRun) {
+			a.logger.Info(fmt.Sprintf("The build pipelineRun %s/%s with pr group %s failed, won't create group snapshot", pipelineRun.Namespace, pipelineRun.Name, prGroup))
+			err := gitops.AnnotateSnapshot(a.context, a.snapshot, gitops.PRGroupCreationAnnotation, fmt.Sprintf("The build pipelineRun %s/%s with pr group %s failed, won't create group snapshot", pipelineRun.Namespace, pipelineRun.Name, prGroup), a.client)
+			if err != nil {
+				return controller.RequeueWithError(err)
+			}
+			return controller.ContinueProcessing()
+		}
+
+		// check if build PLR has component snapshot created except the build that snapshot is created from because the build plr has not been labeled with snapshot name
+		if !metadata.HasAnnotation(&pipelineRun, tekton.SnapshotNameLabel) && !metadata.HasLabelWithValue(a.snapshot, gitops.BuildPipelineRunNameLabel, pipelineRun.Name) {
+			a.logger.Info(fmt.Sprintf("The build pipelineRun %s/%s with pr group %s has succeeded but component snapshot has not been created now", pipelineRun.Namespace, pipelineRun.Name, prGroup))
+			return controller.ContinueProcessing()
+		}
+	}
+
+	groupSnapshot, componentSnapshotInfos, err := a.prepareGroupSnapshot(a.application, prGroupHash)
+	if err != nil {
+		a.logger.Error(err, "Failed to prepare group snapshot")
+		return controller.RequeueWithError(err)
+	}
+
+	if componentSnapshotInfos == nil || len(componentSnapshotInfos) < 2 {
```

Comment:
Is this check needed if the number of `componentSnapshotInfos` is also checked [here](https://github.com/konflux-ci/integration-service/pull/842/files#diff-cbf7e6d58cc12a67b910b5b9108ba68d206e75cc8560ca0ab2b19e2d34415702R914)?

Reply:
Updated. Thanks!